vip-mail.php

```diff
@@ -8,6 +8,19 @@
 License: GPL version 2 or later - http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 */
 
+/**
+ * No-Op wp_mail
+ *
+ * If VIP_BLOCK_WP_MAIL is set, we want to ensure wp_mail is not
+ * sending any emails. Since wp_mail is a pluggable function, we
+ * can redefine it here to do nothing.
+ */
+if ( defined( 'WPCOM_VIP_BLOCK_WP_MAIL' ) && WPCOM_VIP_BLOCK_WP_MAIL && ! function_exists( 'wp_mail' ) ) {
+	function wp_mail( $to, $subject, $message, $headers = '', $attachments = array() ) {
```

Comment:
> It's probably helpful to log a notice when we skip an email (in case we need to debug things).

Can do. We're a little limited in what we can capture because the override point for the phpmailer class is after all the information about the email [has been set](https://github.com/WordPress/WordPress/blob/master/wp-includes/pluggable.php#L477-L488) and some things like the [To address](https://github.com/WordPress/WordPress/blob/master/wp-includes/class-phpmailer.php#L476-L481) are protected variables.

Reply:
`$phpmailer->getAllRecipientAddresses()` should return the `To`/`cc`/etc.

Reply:
Added in b65f1c8